src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+    this.checkIfAlive();
+  }
+
+  checkIfAlive() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((a) => a.health > 0);
```

Comment:
The `checkIfAlive` method modifies the static `Animal.alive` array directly by filtering out animals with health less than or equal to zero. This approach can lead to issues if multiple instances are created and checked for health, as it modifies the shared static array for all instances. Consider a different approach to handle the removal of dead animals, such as maintaining a separate list of dead animals or using a different mechanism to update the `alive` array.